README.md

```diff
@@ -1,2 +1,422 @@
+    // Because the type is passed in as a string literal, this code does not need to
+    // have a dependency on the module that defines `ChildView`. All that is required
+    // for this code to compile is for there to be an
+    // `@Instantiable public struct ChildView: View` in the codebase.
+    @Instantiated(fulfilledByType: "ChildView")
+    private let childViewBuilder: Instantiator<some View>
+}
+```
+
+### @Forwarded
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Forwarded` macro](Sources/SafeDI/PropertyDecoration/Forwarded.swift) are forwarded into the SafeDI dependency tree by a [`ForwardingInstantiator`](Sources/SafeDI/DelayedInstantiation/ForwardingInstantiator.swift) instance’s `instantiate(…)` method. A `@Forwarded`-decorated property is available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
+
+A single `@Instantiable` type may have at most one `@Forwarded`-decorated property.
+
+### @Received
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Received` macro](Sources/SafeDI/PropertyDecoration/Received.swift) are injected into the enclosing type‘s initializer. Received properties must be [`@Instantiated`](#instantiated) or [`@Forwarded`](#forwarded) by an object higher up in the dependency tree.
+
+### Delayed instantiation
+
+When you want to instantiate a dependency after your `init(…)`, you need to declare an `Instantiator<Dependency>`-typed property as `@Instantiated` or `@Received`.
+
+#### Instantiator
+
+The `Instantiator` type is how SafeDI enables deferred instantiation of an `@Instantiable` type. `Instantiator` has a single generic that matches the type of the to-be-instantiated instance. Creating an `Instantiator` property is as simple as creating any other property in the SafeDI ecosystem:
+
+```swift
+@Instantiable
+public struct MyApp: App {
+    public var body: some Scene {
+        WindowGroup {
+            // Returns a new instance of a `ContentView`.
+            contentViewInstantiator.instantiate()
+        }
+    }
+
+    public init(contentViewInstantiator: Instantiator<ContentView>) {
+        self.contentViewInstantiator = contentViewInstantiator
+    }
+
+    /// A private property that knows how to instantiate a content view.
+    @Instantiated
+    private let contentViewInstantiator: Instantiator<ContentView>
+}
+```
+
+It is possible to write a `Instantiator` with a type-erased generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+#### ForwardingInstantiator
+
+The `ForwardingInstantiator` type is how SafeDI enables instantiating any `@Instantiable` type with a `@Forwarded` property. `ForwardingInstantiator` has two generics. The first generic must match the type of the `@Forwarded` property. The second generic matches the type of the to-be-instantiated instance.
+
+```swift
```

Comment:
Can we add another simple example in UIKit for an App level injection as well?

Reply:
We can, but honestly I have significantly less-strong opinions about the right way to define a root using SafeDI in UIKit-world. 

Because `AppDelegate` and `SceneDelegate` must be separate objects, and neither object has a good way to reference the other (outside of the scene delegate referencing and casting `UIApplication.shared.delegate`)... there's no _great_ way to make an application root like there is in `SwiftUI`.

In my own spikes I've been resorting to making a
```
@Instantiable
public final class Root {

    public init(...) { ... }

    static let shared = Root() // uses initializer generated by SafeDI

    // Instantiated properties here
    ...
}
```

And then utilizing `Root.shared` in both the `AppDelegate` and `SceneDelegate`. This is a pretty kludgy (but workable) approach, and is similar to how folk tend to share objects between these two types anyways. Alternatively, you could have separate dependency trees for your `AppDelegate` and `SceneDelegate`... but that seems like a recipe for pain since scenes and the app do need to share context.

So... yeah. I didn't want to have an opinion on this in the README, since there is no great option. Does that make sense?

I agree that having a UIKit example will aid folk who are adopting this system in UIKit, but also... it's not hard to make the leap from "how do I share dependencies between these two delegates" to "I should create a self-vending singleton".